.github/workflows/pr.yml

```diff
@@ -4,11 +4,17 @@ on:
   pull_request:
 
 jobs:
-  geth:
+  amd64:
```

Comment:
I think, we can update the following:
- The job names are in sync with the actual arch: `arm64` --> `linux_arm64`
- The `matrix` entries can remove the redundant values for `arch` and the `platform` (per job) can be hardcoded instead
- Update `actions/checkout@v3` to `actions/checkout@v4`
- Update `docker/build-push-action@v4` to `docker/build-push-action@v6`

The suggestions end up modifying the workflow definition as below:
```
name: Pull Request

on:
  pull_request:

jobs:
  linux_amd64:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        include:
          - file: geth/Dockerfile
            features: 
          - file: reth/Dockerfile
            features: jemalloc,asm-keccak,optimism
    steps:
      - name: Checkout
        uses: actions/checkout@v4
        with:
          ref: ${{ github.event.pull_request.head.sha }}
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v3
      - name: Build the Docker image
        uses: docker/build-push-action@v6
        with:
          context: .
          file: ${{ matrix.file }}
          push: false
          build-args: |
            FEATURES=${{ matrix.features }}
          platforms: linux/amd64

  linux_arm64:
    runs-on: linux-arm64
    strategy:
      matrix:
        include:
          - file: geth/Dockerfile
            features: 
          - file: reth/Dockerfile
            features: jemalloc,optimism
    steps:
      - name: Checkout
        uses: actions/checkout@v4
        with:
          ref: ${{ github.event.pull_request.head.sha }}
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v3
      - name: Build the Docker image
        uses: docker/build-push-action@v6
        with:
          context: .
          file: ${{ matrix.file }}
          push: false
          build-args: |
            FEATURES=${{ matrix.features }}
          platforms: linux/arm64
```